mlfinlab/tests/test_risk_estimators.py

```diff
@@ -0,0 +1,232 @@
+        """
+
+        risk_estimators = RiskEstimators()
+
+        # Covariance matrix
+        cov_matrix = np.array([[0.01, 0.002, -0.001],
+                               [0.002, 0.04, -0.006],
+                               [-0.001, -0.006, 0.01]])
+
+        # Expected correlation matrix
+        expected_matrix = np.array([[1, 0.1, -0.1],
+                                    [0.1, 1, -0.3],
+                                    [-0.1, -0.3, 1]])
+
+        # Finding the covariance matrix
+        corr_matrix = risk_estimators.cov_to_corr(cov_matrix)
+
+        # Testing the first row of the matrix
+        np.testing.assert_almost_equal(corr_matrix, expected_matrix, decimal=5)
+
+    @staticmethod
+    def test_get_pca():
+        """
+        Test the calculation of eigenvalues and eigenvectors from a Hermitian matrix.
+        """
+
+        risk_estimators = RiskEstimators()
+
+        # Correlation matrix as an input
+        corr_matrix = np.array([[1, 0.1, -0.1],
+                                [0.1, 1, -0.3],
+                                [-0.1, -0.3, 1]])
+
+        # Expected correlation matrix
+        expected_eigenvalues = np.array([[1.3562, 0, 0],
+                                         [0, 0.9438, 0],
+                                         [0, 0, 0.7]])
+        first_eigenvector = np.array([-3.69048184e-01, -9.29410263e-01, 1.10397126e-16])
+
+        # Finding the eigenvalues
+        eigenvalues, eigenvectors = risk_estimators._get_pca(corr_matrix)
+
+        # Testing eigenvalues and the first eigenvector
+        np.testing.assert_almost_equal(eigenvalues, expected_eigenvalues, decimal=4)
+        np.testing.assert_almost_equal(eigenvectors[0], first_eigenvector, decimal=5)
+
+    @staticmethod
+    def test_denoised_corr():
+        """
+        Test the shrinkage the eigenvalues associated with noise.
+        """
+
+        risk_estimators = RiskEstimators()
+
+        # Eigenvalues and eigenvectors to use
+        eigenvalues = np.array([[1.3562, 0, 0],
+                                [0, 0.9438, 0],
+                                [0, 0, 0.7]])
+        eigenvectors = np.array([[-3.69048184e-01, -9.29410263e-01, 1.10397126e-16],
+                                 [-6.57192300e-01, 2.60956474e-01, 7.07106781e-01],
+                                 [6.57192300e-01, -2.60956474e-01, 7.07106781e-01]])
+
+        # Expected correlation matrix
+        expected_corr = np.array([[1, 0.13353165, -0.13353165],
+                                  [0.13353165, 1, -0.21921986],
+                                  [-0.13353165, -0.21921986, 1]])
+
+        # Finding the eigenvalues
+        corr_matrix = risk_estimators._denoised_corr(eigenvalues, eigenvectors, 1)
+
+        # Testing if the de-noised correlation matrix is right
+        np.testing.assert_almost_equal(corr_matrix, expected_corr, decimal=4)
+
+    @staticmethod
+    def de_noised_cov():
```

Comment:
Change the function name to - "test_denoise_covariance()"

Reply:
Done